packages/firestore/src/api/timestamp.ts

```diff
@@ -19,6 +19,13 @@ import { Code, FirestoreError } from '../util/error';
 import { primitiveComparator } from '../util/misc';
 
 export class Timestamp {
+  // Midnight at the beginning of 1/1/1 is the earliest Firestore supports.
+  private static readonly MIN_SECONDS = -62135596800;
+  // This will break in the year 10,000.
+  private static readonly MAX_SECONDS = 253402300799;
+  private static readonly MIN_NANOSECONDS = 0;
+  private static readonly MAX_NANOSECONDS = 1e9 - 1;
```

Comment:
Note that "private" is not a thing in the final JavaScript build. When TypeScript transpiles this class to JS, these static members essentially become public. To indicate that we want these member to be treated as private members, we generally prefix all private members of our API classes.

Reply:
FWIW, you could declare these as constants within the file rather than the class and then they wouldn't be exported at all.

Reply:
I have moved these variables and the normalizeAndPad() private method to outside the class declaration so that they will be truly private.